File: backend/src/main/kotlin/com/example/backend/dtos/EnRayonDto.kt
package com.example.backend.dtos

import org.springframework.data.domain.Page
import java.time.LocalDateTime

data class ProduitEnRayonDto(
    val enRayonId: Long? = null, // Optional for adding new products
    val produitId: Long,
    val fournisseurId: Long?,
    val rayonId: Long?,
    val quantite: Int,
    val quantiteRestante: Int,
    val reduction: Int,
    val prixVente: Int? = null,
    val prixAchat: Int? = null,
    val datePeremption: LocalDateTime?=LocalDateTime.now(),
    val dateLivraison: LocalDateTime?=LocalDateTime.now()
)
data class ProduitDetailIncrementEnRayonDto(
    val enRayonId: String? = null,
    val produitDetailId: String? = null,
)

data class EnRayonDto (
   val enRayonId:String? = null,
   val prixAchat:Int? = 0,
   val prixVente :Int? = 0,
   val reductionMax :Int? = 0,
   val quantiteRestante :Int? = 0,
   val datePeremption: String? = null // Getters and Setters
)

data class EnRayonPageableCustomDto(
  var content: Page<Map<String, Any?>?>,
  var totalElements: Long?,
  var totalPages: Int?,
  var pageSize: Int?,
  var pageNumber: Int?,
  var totalAmountEnRayon: Double?,
  var totalQte: Int?,
)

File: backend_py/pharmaPython/app/schemas/enrayon_dto.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime

class ProduitEnRayonDto(BaseModel):
  enRayonId: Optional[int] = None
  produitId: int
  fournisseurId: Optional[int] = None
  rayonId: Optional[int] = None
  quantite: int
  quantiteRestante: int
  reduction: int
  prixVente: Optional[int] = None
  prixAchat: Optional[int] = None
  datePeremption: Optional[datetime] = None
  dateLivraison: Optional[datetime] = None

class ProduitDetailIncrementEnRayonDto(BaseModel):
  enRayonId: Optional[str] = None
  produitDetailId: Optional[str] = None

class EnRayonDto(BaseModel):
  enRayonId: Optional[str] = None
  prixAchat: Optional[int] = 0
  prixVente: Optional[int] = 0
  reductionMax: Optional[int] = 0
  quantiteRestante: Optional[int] = 0
  datePeremption: Optional[str] = None

class EnRayonPageableCustomDto(BaseModel):
  # En Kotlin: Page<Map<String, Any?>> → ici, liste de dicts + métadonnées
  content: List[Dict[str, Any]] = []
  totalElements: Optional[int] = None
  totalPages: Optional[int] = None
  pageSize: Optional[int] = None
  pageNumber: Optional[int] = None
  totalAmountEnRayon: Optional[int] = None
  totalQte: Optional[int] = None
  data: Optional[Dict[str, Any]] = None
